nodejs/awsx/apigateway/api.ts

```diff
@@ -285,9 +285,16 @@ export interface APIArgs {
     /**
      * Routes to use to initialize the APIGateway.
      *
+     * Note: `routes` is allowed to be an `Output<Route[]>`, not just an explicit `Route[]`.
+     * However, there are restrictions if this is passed in as an `Output`. Specifically, none of
+     * the routes passed in are allowed to cause specific resources to be created just for them.
+     *
+     * Because of this, the only routes supported when passing in an `Output<Route[]>` are currently
+     * `IntegrationRoute`s and `RawDataRoute`s.
+     *
```

Comment:
This feels a little precarious - theres a new sharp edge here.

I wonder - could we alleviate this by doing something like accepting an `additionalRoutes` property that can be an` Input` but accepts a subset of the kinds of routes?  That would remove the sharp edge - it would be possible to do both.  That may not be the cleanest API, but something like that feels like it would in principle work and support more scenarios?

Reply:
i would be fine with that.